905.py

```diff
@@ -0,0 +1,15 @@
+class Solution:
+    def sortArrayByParity(self, A: List[int]) -> List[int]:
+        i, j = 0, len(A) - 1
+
+        while i < j:
+            if A[i] % 2 > A[j] % 2:
+                A[i], A[j] = A[j], A[i]
+
+            if A[i] % 2 == 0:
```

Comment:
It seems that you are more comfortable incrementing indexes one position at a time. I like to increment them all at once as long as the loop invariant still holds true.

```python
while l < r and A[l] % 2 == 0:
    l += 1
    
while l < r and A[r] % 2 != 0:
    r -= 1
```

There is no difference in terms of time complexity or performance; it's more of a coding style issue.